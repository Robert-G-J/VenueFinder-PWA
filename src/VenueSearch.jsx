import React, { Component } from "react";

const foursquare = require("react-foursquare")({
  clientID: "A150E11L5JS0WTGP0C1F5E4HAJHKDICLYKQIQTDGX4K21LLE",
  clientSecret: "UEYU0RJNPXDXLDY5FNF1DTSZJ4HO3CUEBTNHTJWDJNB1WLPX"
});

class VenueSearch extends Component {
  constructor(props) {
    super(props);
    // state declaration
    this.state = {
      queryParams: {
        ll: "37.7749, -122.4194",
        query: null,
        v: "20170801"
      },
      queryResponse: []
    };

    // bindings
    this.onChangeSearchField = this.onChangeSearchField.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  onChangeSearchField(e) {
    this.setState({ query: e.target.value });
  }

  handleClick(e) {
    e.preventDefault();
    foursquare.venues.getVenues(this.state.queryParams).then(res => {
      console.log("FSQ response", res);
      this.setState({ queryResponse: res.response.venues });
      console.log("FSQ Response set in state");
    });
  }

  render() {
    return (
      <div className="search__input">
        <h1>VenueSearch Component</h1>
        <div>
          <label />
          <input type="text" onChange={this.onChangeSearchField} />
          <button onClick={this.handleClick}>Search</button>
        </div>
        {/* <div>{this.state.queryResponse && this.state.queryResponse.map()}</div> */}
      </div>
    );
  }
}
export default VenueSearch;
