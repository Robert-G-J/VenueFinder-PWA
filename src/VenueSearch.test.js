import React from "react";
import VenueSearch from "./VenueSearch";
import renderer from "react-test-renderer";

describe("The search component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<VenueSearch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
