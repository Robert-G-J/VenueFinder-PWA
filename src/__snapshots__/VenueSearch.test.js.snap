// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The search component renders correctly 1`] = `
<div
  className="search__input"
>
  <h1>
    VenueSearch Component
  </h1>
  <div>
    <label />
    <input
      onChange={[Function]}
      type="text"
    />
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
